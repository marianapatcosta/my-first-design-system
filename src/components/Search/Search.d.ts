import React, { ChangeEvent, CSSProperties, MouseEventHandler } from 'react';
export interface SearchProps {
    /**
     * element id
     */
    id?: string;
    /**
     * additional css class
     */
    className?: string;
    /**
     * additional style
     */
    style?: CSSProperties;
    /**
     * Search label
     */
    label?: string;
    /**
     * Search name
     */
    name?: string;
    /**
     * Search value
     */
    value: string;
    /**
     * Search's displaying text when no option is selected
     */
    placeholder?: string;
    /**
     * what is maxLength allowed for the search field?
     */
    maxLength?: number;
    /**
     * Is Search disabled?
     */
    disabled?: boolean;
    /**
     * Is Search expansible i.e. is search collapsed and search field is rendered after click,
     * or search field is always visible?
     */
    isExpandable?: boolean;
    /**
     * Change handler
     */
    onChange: (event: ChangeEvent<HTMLInputElement>) => void;
    /**
     * Click handler to remove the value in search field
     */
    onClear: MouseEventHandler<HTMLImageElement>;
}
export declare const Search: React.FC<SearchProps>;
