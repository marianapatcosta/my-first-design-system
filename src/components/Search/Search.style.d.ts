export declare const StyledSearchWrapper: import("styled-components").StyledComponent<"div", any, {}, never>;
export declare const StyledLabel: import("styled-components").StyledComponent<"label", any, {
    disabled?: boolean | undefined;
}, never>;
export declare const StyledSearchContainer: import("styled-components").StyledComponent<"div", any, {
    isExpanded?: boolean | undefined;
}, never>;
export declare const StyledSearch: import("styled-components").StyledComponent<"input", any, {
    disabled?: boolean | undefined;
    isExpanded?: boolean | undefined;
}, never>;
export declare const StyledSearchIconWrapper: import("styled-components").StyledComponent<"div", any, {
    isExpanded?: boolean | undefined;
    isExpandable?: boolean | undefined;
}, never>;
export declare const StyledSearchIcon: import("styled-components").StyledComponent<"img", any, {
    isExpanded?: boolean | undefined;
    isExpandable?: boolean | undefined;
    hasValue?: boolean | undefined;
}, never>;
