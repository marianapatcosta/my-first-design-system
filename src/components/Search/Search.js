var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
import { jsx as _jsx, jsxs as _jsxs } from "react/jsx-runtime";
import { useEffect, useState, } from 'react';
import { Close, MagnifyingGlass } from '../../assets/icons';
import { isEventValid } from '../../utils/utils';
import { StyledSearchWrapper, StyledLabel, StyledSearchContainer, StyledSearch, StyledSearchIconWrapper, StyledSearchIcon, } from './Search.style';
export var Search = function (_a) {
    var className = _a.className, id = _a.id, label = _a.label, value = _a.value, disabled = _a.disabled, placeholder = _a.placeholder, isExpandable = _a.isExpandable, onClear = _a.onClear, props = __rest(_a, ["className", "id", "label", "value", "disabled", "placeholder", "isExpandable", "onClear"]);
    var _b = useState(false), isExpanded = _b[0], setIsExpanded = _b[1];
    useEffect(function () { return setIsExpanded(isExpandable ? false : true); }, [isExpandable]);
    var toggleIsExpanded = function () {
        return isExpandable && setIsExpanded(function (prevIsExpanded) { return !prevIsExpanded; });
    };
    var handleClear = function (event) {
        if (isExpandable)
            setIsExpanded(false);
        onClear(event);
    };
    var handleKeyDown = function (event) {
        return isEventValid(event) && isExpandable && toggleIsExpanded();
    };
    return (_jsxs(StyledSearchWrapper, __assign({ className: className }, { children: [label && (_jsx(StyledLabel, __assign({ htmlFor: id, disabled: disabled }, { children: label }), void 0)),
            _jsxs(StyledSearchContainer, __assign({ isExpanded: !isExpandable ? true : isExpanded }, { children: [_jsx(StyledSearch, __assign({ "data-testid": 'search-field', id: id, "aria-label": !!label ? label : placeholder, type: 'search', placeholder: isExpanded ? placeholder : '', disabled: disabled, isExpanded: !isExpandable ? true : isExpanded, value: isExpanded ? value : '', onBlur: function () { return setIsExpanded(false); }, onKeyDown: handleKeyDown }, props), void 0),
                    _jsx(StyledSearchIconWrapper, { children: _jsx(StyledSearchIcon, { "data-testid": 'search-icon', alt: !!value && isExpanded ? 'clear-icon' : 'magnifying-glass-icon', src: !!value && isExpanded ? Close : MagnifyingGlass, isExpandable: isExpandable, hasValue: !!value, isExpanded: !isExpandable ? true : isExpanded, onClick: !value ? toggleIsExpanded : handleClear }, void 0) }, void 0)] }), void 0)] }), void 0));
};
Search.defaultProps = {
    id: '',
    className: '',
    style: {},
    label: '',
    name: '',
    maxLength: 150,
    placeholder: 'Search',
    disabled: false,
    isExpandable: false,
    value: '',
    onChange: function () { return null; },
    onClear: function () { return null; },
};
