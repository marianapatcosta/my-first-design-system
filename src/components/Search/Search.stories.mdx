import { Meta } from "@storybook/addon-docs/blocks";
import { Search } from "./Search";

<Meta title="Design System|Search" component={Search} />

# Search
use search field when it is supposed that the user provides some written information for searching purposes

## Usage
- Default Search:
  <Story name='Default'>
    <Search value='hello' />
  </Story>

- Expansible Search:
  <Story name='Expansible'>
    <Search value='hello' isExpandable={true}/>
  </Story>

- Search With Label:
  <Story name='WithLabel'>
    <Search value='hello' label='Search'/>
  </Story>

## Other important notes
not applicable.