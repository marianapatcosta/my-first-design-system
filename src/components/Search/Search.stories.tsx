import React from 'react';
import { Story, Meta } from '@storybook/react/types-6-0';
import { Search, SearchProps } from './Search';

export default {
  title: 'Search',
  component: Search,
  // args behavior in each story
  args: {},
} as Meta;

const Template: Story<SearchProps> = (args) => <Search {...args} />;

export const Default = Template.bind({});
Default.args = { value: 'hello' };

export const Expansible = Template.bind({});
Expansible.args = { isExpandable: true };

export const WithLabel = Template.bind({});
WithLabel.args = { label: 'Search' };
