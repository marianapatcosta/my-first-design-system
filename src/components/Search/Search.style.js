var __makeTemplateObject = (this && this.__makeTemplateObject) || function (cooked, raw) {
    if (Object.defineProperty) { Object.defineProperty(cooked, "raw", { value: raw }); } else { cooked.raw = raw; }
    return cooked;
};
import styled from 'styled-components';
import { baseStyles, disabledStyles, colors, typography } from '../../styles';
export var StyledSearchWrapper = styled.div(templateObject_1 || (templateObject_1 = __makeTemplateObject(["\n  ", "\n  margin: 1rem 0;\n  width: 100%;\n  max-width: 15rem;\n"], ["\n  ", "\n  margin: 1rem 0;\n  width: 100%;\n  max-width: 15rem;\n"])), baseStyles);
export var StyledLabel = styled.label(templateObject_2 || (templateObject_2 = __makeTemplateObject(["\n  display: block;\n  color: ", ";\n  margin-bottom: 0.3rem;\n  margin-left: 0.3rem;\n  ", "\n"], ["\n  display: block;\n  color: ", ";\n  margin-bottom: 0.3rem;\n  margin-left: 0.3rem;\n  ",
    "\n"])), colors.highlight, function (_a) {
    var disabled = _a.disabled;
    return disabled && disabledStyles + "; color: " + colors.disabled;
});
export var StyledSearchContainer = styled.div(templateObject_3 || (templateObject_3 = __makeTemplateObject(["\n  display: flex;\n  margin-left: auto;\n  box-shadow: 0 0.063rem 0.25rem ", ";\n  transition: width 0.3s ease-in-out;\n  box-shadow: 0 0.063rem 0.25rem ", ";\n  width: 100%;\n  margin-left: auto;\n  background-color: ", ";\n\n  ", "\n"], ["\n  display: flex;\n  margin-left: auto;\n  box-shadow: 0 0.063rem 0.25rem ", ";\n  transition: width 0.3s ease-in-out;\n  box-shadow: 0 0.063rem 0.25rem ", ";\n  width: 100%;\n  margin-left: auto;\n  background-color: ", ";\n\n  ", "\n"])), colors.shadow, colors.shadow, colors.primary, function (_a) {
    var isExpanded = _a.isExpanded;
    return !isExpanded && "width: 2rem !important;";
});
export var StyledSearch = styled.input(templateObject_4 || (templateObject_4 = __makeTemplateObject(["\n  color: ", ";\n  font-family: ", ";\n  font-size: 95%;\n  width: 100%;\n  border: none;\n  box-sizing: border-box;\n  padding: 0.3rem 0.625rem;\n  position: relative;\n  display: block;\n  height: 2rem;\n  background-color: ", ";\n\n  :-webkit-autofill,\n  :-webkit-autofill:hover,\n  :-webkit-autofill:focus,\n  :-webkit-autofill:active {\n    -webkit-text-fill-color: ", ";\n    -webkit-box-shadow: 0 0 0 2rem ", " inset !important;\n  }\n\n  /* clears the default 'X' from Internet Explorer */\n  ::-ms-clear {\n    display: none;\n    width: 0;\n    height: 0;\n  }\n  ::-ms-reveal {\n    display: none;\n    width: 0;\n    height: 0;\n  }\n\n  /* clears the default 'X' from Chrome */\n  ::-webkit-search-decoration,\n  ::-webkit-search-cancel-button,\n  ::-webkit-search-results-button,\n  ::-webkit-search-results-decoration {\n    display: none;\n  }\n\n  ", ";\n\n  ", ";\n"], ["\n  color: ", ";\n  font-family: ", ";\n  font-size: 95%;\n  width: 100%;\n  border: none;\n  box-sizing: border-box;\n  padding: 0.3rem 0.625rem;\n  position: relative;\n  display: block;\n  height: 2rem;\n  background-color: ", ";\n\n  :-webkit-autofill,\n  :-webkit-autofill:hover,\n  :-webkit-autofill:focus,\n  :-webkit-autofill:active {\n    -webkit-text-fill-color: ", ";\n    -webkit-box-shadow: 0 0 0 2rem ", " inset !important;\n  }\n\n  /* clears the default 'X' from Internet Explorer */\n  ::-ms-clear {\n    display: none;\n    width: 0;\n    height: 0;\n  }\n  ::-ms-reveal {\n    display: none;\n    width: 0;\n    height: 0;\n  }\n\n  /* clears the default 'X' from Chrome */\n  ::-webkit-search-decoration,\n  ::-webkit-search-cancel-button,\n  ::-webkit-search-results-button,\n  ::-webkit-search-results-decoration {\n    display: none;\n  }\n\n  ", ";\n\n  ",
    ";\n"])), colors.font, typography.type.body, colors.primary, colors.font, colors.primary, function (_a) {
    var disabled = _a.disabled;
    return disabled && " " + disabledStyles;
}, function (_a) {
    var isExpanded = _a.isExpanded;
    return !isExpanded ? "display: none; caret-color: transparent;" : '';
});
export var StyledSearchIconWrapper = styled.div(templateObject_5 || (templateObject_5 = __makeTemplateObject(["\n  position: relative;\n  width: 2rem;\n  height: 2rem;\n  background-color: ", ";\n  ", "\n"], ["\n  position: relative;\n  width: 2rem;\n  height: 2rem;\n  background-color: ", ";\n  ",
    "\n"])), colors.primary, function (_a) {
    var isExpandable = _a.isExpandable, isExpanded = _a.isExpanded;
    return isExpandable && !isExpanded && "cursor: pointer;";
});
export var StyledSearchIcon = styled.img(templateObject_6 || (templateObject_6 = __makeTemplateObject(["\n  position: absolute;\n  width: 1rem;\n  height: 1rem;\n  top: 0.5rem;\n  right: 0.625rem;\n  ", ";\n\n  ", "\n  ", "\n\n  ", "\n"], ["\n  position: absolute;\n  width: 1rem;\n  height: 1rem;\n  top: 0.5rem;\n  right: 0.625rem;\n  ", ";\n\n  ", "\n  ", "\n\n  ",
    "\n"])), colors.icon, function (_a) {
    var hasValue = _a.hasValue;
    return hasValue && "cursor: pointer;";
}, function (_a) {
    var isExpandable = _a.isExpandable;
    return isExpandable && "cursor: pointer;";
}, function (_a) {
    var isExpanded = _a.isExpanded;
    return !isExpanded &&
        "top: 0;\n      bottom: 0;\n      margin: auto;";
});
var templateObject_1, templateObject_2, templateObject_3, templateObject_4, templateObject_5, templateObject_6;
