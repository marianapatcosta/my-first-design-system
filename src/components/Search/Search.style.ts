import styled from 'styled-components';
import { baseStyles, disabledStyles, colors, typography } from '../../styles';

export const StyledSearchWrapper = styled.div`
  ${baseStyles}
  margin: 1rem 0;
  width: 100%;
  max-width: 15rem;
`;

export const StyledLabel = styled.label<{ disabled?: boolean }>`
  display: block;
  color: ${colors.highlight};
  margin-bottom: 0.3rem;
  margin-left: 0.3rem;
  ${({ disabled }) =>
    disabled && `${disabledStyles}; color: ${colors.disabled}`}
`;

export const StyledSearchContainer = styled.div<{ isExpanded?: boolean }>`
  display: flex;
  margin-left: auto;
  box-shadow: 0 0.063rem 0.25rem ${colors.shadow};
  transition: width 0.3s ease-in-out;
  box-shadow: 0 0.063rem 0.25rem ${colors.shadow};
  width: 100%;
  margin-left: auto;
  background-color: ${colors.primary};

  ${({ isExpanded }) => !isExpanded && `width: 2rem !important;`}
`;

export const StyledSearch = styled.input<{
  disabled?: boolean;
  isExpanded?: boolean;
}>`
  color: ${colors.font};
  font-family: ${typography.type.body};
  font-size: 95%;
  width: 100%;
  border: none;
  box-sizing: border-box;
  padding: 0.3rem 0.625rem;
  position: relative;
  display: block;
  height: 2rem;
  background-color: ${colors.primary};

  :-webkit-autofill,
  :-webkit-autofill:hover,
  :-webkit-autofill:focus,
  :-webkit-autofill:active {
    -webkit-text-fill-color: ${colors.font};
    -webkit-box-shadow: 0 0 0 2rem ${colors.primary} inset !important;
  }

  /* clears the default 'X' from Internet Explorer */
  ::-ms-clear {
    display: none;
    width: 0;
    height: 0;
  }
  ::-ms-reveal {
    display: none;
    width: 0;
    height: 0;
  }

  /* clears the default 'X' from Chrome */
  ::-webkit-search-decoration,
  ::-webkit-search-cancel-button,
  ::-webkit-search-results-button,
  ::-webkit-search-results-decoration {
    display: none;
  }

  ${({ disabled }) => disabled && ` ${disabledStyles}`};

  ${({ isExpanded }) =>
    !isExpanded ? `display: none; caret-color: transparent;` : ''};
`;

export const StyledSearchIconWrapper = styled.div<{
  isExpanded?: boolean;
  isExpandable?: boolean;
}>`
  position: relative;
  width: 2rem;
  height: 2rem;
  background-color: ${colors.primary};
  ${({ isExpandable, isExpanded }) =>
    isExpandable && !isExpanded && `cursor: pointer;`}
`;

export const StyledSearchIcon = styled.img<{
  isExpanded?: boolean;
  isExpandable?: boolean;
  hasValue?: boolean;
}>`
  position: absolute;
  width: 1rem;
  height: 1rem;
  top: 0.5rem;
  right: 0.625rem;
  ${colors.icon};

  ${({ hasValue }) => hasValue && `cursor: pointer;`}
  ${({ isExpandable }) => isExpandable && `cursor: pointer;`}

  ${({ isExpanded }) =>
    !isExpanded &&
    `top: 0;
      bottom: 0;
      margin: auto;`}
`;
