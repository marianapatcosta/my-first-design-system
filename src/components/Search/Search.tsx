import React, {
  ChangeEvent,
  CSSProperties,
  KeyboardEvent,
  MouseEvent,
  MouseEventHandler,
  useEffect,
  useState,
} from 'react';
import { Close, MagnifyingGlass } from '../../assets/icons';
import { isEventValid } from '../../utils/utils';
import {
  StyledSearchWrapper,
  StyledLabel,
  StyledSearchContainer,
  StyledSearch,
  StyledSearchIconWrapper,
  StyledSearchIcon,
} from './Search.style';

export interface SearchProps {
  /**
   * element id
   */
  id?: string;
  /**
   * additional css class
   */
  className?: string;
  /**
   * additional style
   */
  style?: CSSProperties;
  /**
   * Search label
   */
  label?: string;
  /**
   * Search name
   */
  name?: string;
  /**
   * Search value
   */
  value: string;
  /**
   * Search's displaying text when no option is selected
   */
  placeholder?: string;
  /**
   * what is maxLength allowed for the search field?
   */
  maxLength?: number;
  /**
   * Is Search disabled?
   */
  disabled?: boolean;
  /**
   * Is Search expansible i.e. is search collapsed and search field is rendered after click,
   * or search field is always visible?
   */
  isExpandable?: boolean;
  /**
   * Change handler
   */
  onChange: (event: ChangeEvent<HTMLInputElement>) => void;
  /**
   * Click handler to remove the value in search field
   */
  onClear: MouseEventHandler<HTMLImageElement>;
}

export const Search: React.FC<SearchProps> = ({
  className,
  id,
  label,
  value,
  disabled,
  placeholder,
  isExpandable,
  onClear,
  ...props
}) => {
  const [isExpanded, setIsExpanded] = useState<boolean>(false);

  useEffect(() => setIsExpanded(isExpandable ? false : true), [isExpandable]);

  const toggleIsExpanded = () =>
    isExpandable && setIsExpanded((prevIsExpanded: boolean) => !prevIsExpanded);

  const handleClear = (event: MouseEvent<HTMLImageElement>) => {
    if (isExpandable) setIsExpanded(false);
    onClear(event);
  };

  const handleKeyDown = (event: KeyboardEvent<HTMLInputElement>) =>
    isEventValid(event) && isExpandable && toggleIsExpanded();

  return (
    <StyledSearchWrapper className={className}>
      {label && (
        <StyledLabel htmlFor={id} disabled={disabled}>
          {label}
        </StyledLabel>
      )}
      <StyledSearchContainer isExpanded={!isExpandable ? true : isExpanded}>
        <StyledSearch
          data-testid='search-field'
          id={id}
          type='search'
          placeholder={isExpanded ? placeholder : ''}
          disabled={disabled}
          isExpanded={!isExpandable ? true : isExpanded}
          value={isExpanded ? value : ''}
          onBlur={() => setIsExpanded(false)}
          onKeyDown={handleKeyDown}
          {...props}
        />
        <StyledSearchIconWrapper>
          <StyledSearchIcon
            data-testid='search-icon'
            alt={value && isExpanded ? 'clear-icon' : 'magnifying-glass-icon'}
            src={value && isExpanded ? Close : MagnifyingGlass}
            isExpandable={isExpandable}
            isExpanded={!isExpandable ? true : isExpanded}
            onClick={!value ? toggleIsExpanded : handleClear}
          />
        </StyledSearchIconWrapper>
      </StyledSearchContainer>
    </StyledSearchWrapper>
  );
};

Search.defaultProps = {
  id: '',
  className: '',
  style: {},
  label: '',
  name: '',
  maxLength: 150,
  placeholder: 'Search',
  disabled: false,
  isExpandable: false,
  value: '',
  onChange: () => null,
  onClear: () => null,
};
