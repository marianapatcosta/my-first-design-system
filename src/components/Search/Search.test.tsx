import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { Search } from './Search';

describe('Search', () => {
  const defaultProps = {
    label: 'Search',
    value: 'Hello',
    onChange: () => null,
    onClear: () => null,
  };

  test('renders without errors and matches snapshot', () => {
    const {
      container: { firstChild },
    } = render(<Search {...defaultProps} />);

    expect(firstChild).toMatchSnapshot();
  });

  test('should handle change events', () => {
    const onChangeMock = jest.fn();
    const { getByTestId } = render(
      <Search {...defaultProps} onChange={onChangeMock} />
    );
    const input = getByTestId('search-field') as HTMLInputElement;
    fireEvent.click(input, { target: { value: 'bye' } });
    expect(input.value).toBe('bye');
  });

  test('should return null on change if onChange prop is not passed', () => {
    const result =
      Search.defaultProps?.onChange && Search.defaultProps.onChange();
    expect(result).toBeNull();
  });

  test('should return null on close icon click if onClear prop is not passed', () => {
    const result =
      Search.defaultProps?.onClear && Search.defaultProps.onClear();
    expect(result).toBeNull();
  });

  test('should display magnifying glass icon when no value is inserted in search field', () => {
    const { getByAltText } = render(<Search {...defaultProps} value='' />);

    const magnifyingGlassIcon = getByAltText('magnifying-glass-icon');
 //   const clearIcon = getByAltText('clear icon');

    expect(magnifyingGlassIcon).toBeTruthy();
  //  expect(clearIcon).toBeFalsy();
  });

  test('should display clear icon when some value is inserted in search field', () => {
    const { getByAltText } = render(<Search {...defaultProps} />);
    const clearIcon = getByAltText('clear-icon');
    clearIcon && fireEvent.click(clearIcon);
    expect(clearIcon).toBeTruthy();
  });

  test('should clear input field after click on clear icon ', () => {
    const onClearMock = jest.fn();
    const { getByAltText, getByTestId } = render(
      <Search {...defaultProps} onClear={onClearMock} />
    );
    const clearIcon = getByAltText('clear-icon');
    clearIcon && fireEvent.click(clearIcon);

    expect(onClearMock).toBeCalled();
  });

  describe('Disabled Search', () => {
    test('should not handle change events', () => {
      const onChangeMock = jest.fn();
      const {
        container: { firstChild },
      } = render(
        <Search {...defaultProps} disabled={true} onChange={onChangeMock} />
      );

      firstChild && fireEvent.click(firstChild);

      expect(onChangeMock).not.toBeCalled();
    });
  });
});
