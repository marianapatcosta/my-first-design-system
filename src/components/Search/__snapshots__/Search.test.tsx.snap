// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders without errors and matches snapshot 1`] = `
<div
  class="sc-bdfBwQ bSfouD"
>
  <label
    class="sc-gsTCUz ecMFjJ"
    for=""
  >
    Search
  </label>
  <div
    class="sc-dlfnbm hoLpfF"
  >
    <input
      class="sc-hKgILt dDwLED"
      data-testid="search-field"
      id=""
      maxlength="150"
      name=""
      placeholder="Search"
      type="search"
      value="Hello"
    />
    <div
      class="sc-eCssSg cUBvld"
    >
      <img
        alt="clear-icon"
        class="sc-jSgupP izDhMj"
        data-testid="search-icon"
        src="close.svg"
      />
    </div>
  </div>
</div>
`;
